jakttest/process.cpp

```diff
@@ -92,7 +92,7 @@ ErrorOr<Optional<i32>> poll_process_exit(i32 pid)
     }
     // NOTE: compiler seems to struggle to find a constructor when
     // nesting error & optional
-    return Optional<i32>(exit_code);
+    return Optional<i32>(WEXITSTATUS(wstatus));
```

Comment:
This isn't sufficient. You need to check WIFEXITED before calling WEXITSTATUS. it's possible the process was signaled (abort, panic, ...) Instead of exiting normally.

Reply:
Hmm, I didn't think we'd care if the test has crashed. Maybe I should make `poll_process_exit` return a `bool` telling whether it was signaled?

Reply:
That is, if you get stopped/continued, you need to keep waiting for that pid until it's *really* dead

Reply:
Enums are hard to get from C++ into Jakt, since Jakt enums work with variants. It's not easy to have transparent types between C++ and Jakt and have that amount of information given. Best I can think of is making the returned code be the signal if `WIFSIGNALED` is set.

Reply:
I'm going to revert all changes on this PR beyond just returning the exit code. The scheduler doesn't care if the process exited normally or abruptly right now, it just wants to know if it's in the exited state so it can fit a new task. If signal handling is deemed necessary later we can just land a new PR.